### PR TITLE
[NEAT-230] 🏗Extending solution model

### PR DESCRIPTION
**Pull request by doctrino**
As always ran into multiple issues when writing the tutorial, resulting in the following code changes

* Created a `.dump()` method instead of hooking into the pydantic `.model_dump`. The reason is that we want to have custom arguments as `as_reference` in the dump method.
* In `InformationConverter`, automatically bump container when `extension=addition/reshape`.
* Bugs in DMS validation. For example, `extension=addition` allows for adding new properties.
* Fix bug in `.as_schema()` when `extension=addition`, as it was not creating a new view with all properties, and the new data model was missing all views from the last sheet.

### CODE REVIEW

File: docs/tutorials/data-modeling-lifecycle/part-4-extending-solution-model.md [hunk 1]
```diff
@@ -1,5 +1,168 @@
...
+Olav selects the following options:
+
+* **Data model id**: This is the ID of the current analytic solution model.
+* **Reference data model id** This is the ID of the enterprise model that the solution model is based.
+* **Report formatter**: This is used in the validation of the model. The current solution model should be valid,
+  so this is likely not needed.
+* **Role**: This is which format Olav wants to download the model. He selects `dms_architect` as he wants the
+  model in the implementation format as he will focus on how data is stored in the model.
+
+Furthermore, he clicks on the `Create Excel Sheet` step which opens a modal with the configuration for the export
```
Comment (nikokaoja) [inline docs/tutorials/data-modeling-lifecycle/part-4-extending-solution-model.md:32]: Where is this step? I cannot find it

Comment (nikokaoja) [reply docs/tutorials/data-modeling-lifecycle/part-4-extending-solution-model.md:32]: I think you mean `RulesToExcel` ?

**Review by nikokaoja — APPROVED** [review]

File: docs/tutorials/data-modeling-lifecycle/part-4-extending-solution-model.md (hunk 1 above)
Comment (doctrino) [reply docs/tutorials/data-modeling-lifecycle/part-4-extending-solution-model.md:32]: No, it is `Create Excel File` in the `Import_DMS` workflow

File: cognite/neat/rules/models/dms/_exporter.py [hunk 2]
```diff
@@ -58,19 +59,83 @@ def __init__(
...
 
+        last_schema: DMSSchema | None = None
+        if self.rules.last:
+            last_schema = self.rules.last.as_schema()
+            # Remove the views that are in the current model, last + current should equal the full model
+            # without any duplicates
+            last_schema.views = ViewApplyDict(
+                {view_id: view for view_id, view in last_schema.views.items() if view_id not in views}
+            )
+            last_schema.containers = ContainerApplyDict(
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/models/dms/_exporter.py:119]: In the code above.

**Pull request merged**